concepts/match-basics/about.md

```diff
@@ -56,66 +64,85 @@ fn add_one(x: Option<u8>) -> Option<u8> {
 
 fn main() {
     let three: Option<u8> = Option::Some(3);
-    let four: Option<u8> = plus_one(three);
-    let none = plus_one(Option::None);
+    let four: Option<u8> = add_one(three);
+    let none = add_one(Option::None);
+
+    println!("{three:?}"); // prints "Option::Some(3)"
+    println!("{four:?}"); // prints "Option::Some(4)"
+    println!("{none:?}"); // prints "Option::None(())"
 }
 ```
 
-## Matches cover all possible cases
+> Note: `{:?}` is a special formatting syntax that allows to print a debug form of the parameter passed to the `println!` macro. You can find more information about it in [The Cairo Book](https://book.cairo-lang.org/appendix-03-derivable-traits.html#debug-for-printing-and-debugging).
 
-One benefit of using the `match` control flow is due to its exhaustive nature. In Cairo, especially in the case of `Option<T>`, it expects that every possible pattern must be exhaustively handled else it will revert at compile time.
+One benefit of using the `match` control flow is its exhaustive nature: the arms’ patterns must cover _all_ possibilities. Consider the version of the previous example that does not handle the `None` case:
 
 ```rust
 fn add_one(x: Option<u8>) -> Option<u8> {
     match x {
         Option::Some(val) => Option::Some(val + 1),
     }
+     // this code does not compile, as we forgot to handle the `None` case
 }
 ```
 
-This kind of bug is easier for the compiler to spot because in this case, the possibility that the `x` value could be a null value was not handled. The revert error looks like this error:
-
-```sh
-$ scarb cairo-run
-   Compiling missing_match_arm v0.1.0 (home/Scarb.toml)
-error: Missing match arm: `None` not covered.
- --> home/src/lib.cairo:5:5
-    match x {
-    ^*******^
-
-error: could not compile `missing_match_arm` due to previous error
-error: `scarb metadata` exited with error
-
-```
-
 The error shows that Cairo is aware of other uncovered possible cases that x could be. This error spells out the possibility of the `x` value being null and protect us from assuming that `x` will never be null.
```

Comment:
```suggestion
```
Remove this sentence as well, as it makes little sense without the previous (now deleted) content